src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,44 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
-}
+  // write your code here0
+  constructor(name, health = 100, hidden = false) {
+    super(name, health, hidden);
+    this.hidden = hidden;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 
+  hide() {
+    this.hidden = true;
+  }
+}
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  bite(enemy) {
+    if (enemy instanceof Herbivore && enemy.hidden === false) {
+      enemy.health -= 50;
+
+      if (enemy.health <= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(enemy), 1);
+      }
+    }
+  }
```

Comment:
[CODE PERFOMANCE] You are using `indexOf` and `splice` methods to remove the dead animal from the `Animal.alive` array. This is not efficient because both methods iterate over the array. Instead, you can use the `filter` method which will iterate over the array only once and will remove the dead animal.